protobuf/src/main/java/io/grpc/protobuf/ProtoInputStream.java

```diff
@@ -70,9 +70,11 @@ public int drainTo(OutputStream target) throws IOException {
       written = message.getSerializedSize();
       message.writeTo(target);
       message = null;
-    } else {
+    } else if (partial != null) {
       written = (int) ByteStreams.copy(partial, target);
       partial = null;
+    } else {
+      written = 0;
```

Comment:
No, I'd rather the compiler help verify that each condition block sets the value.